pdr_backend/lake/etl.py

```diff
@@ -241,78 +302,31 @@ def _calc_bronze_start_end_ts(self) -> Tuple[UnixTimeMs, UnixTimeMs]:
 
         return from_timestamp, to_timestamp
 
-    @enforce_types
-    def create_etl_view(self, table_name: str):
-        # Assemble view query and create the view
-        assert (
-            table_name in _ETL_REGISTERED_TABLE_NAMES
-        ), f"{table_name} must be a bronze table"
-
-        db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
-        temp_table = TempTable(table_name)
-        etl_view = ETLTable(table_name)
-
-        table_exists = db.table_exists(table_name)
-        temp_table_exists = db.table_exists(temp_table.fullname)
-        etl_view_exists = db.view_exists(etl_view.fullname)
-        assert temp_table_exists, f"{temp_table.fullname} must already exist"
-        if etl_view_exists:
-            logger.error("%s must not exist", etl_view.fullname)
-            return
-
-        view_query = None
-
-        if table_exists and temp_table_exists:
-            view_query = """
-                CREATE VIEW {} AS
-                (
-                    SELECT * FROM {}
-                    UNION ALL
-                    SELECT * FROM {}
-                )""".format(
-                etl_view.fullname,
-                table_name,
-                temp_table.fullname,
-            )
-            db.query_data(view_query)
-            logger.info(
-                "  Created %s view using %s table and %s temp table",
-                etl_view.fullname,
-                table_name,
-                temp_table.fullname,
-            )
-        else:
-            view_query = f"CREATE VIEW {etl_view.fullname} AS SELECT * FROM {temp_table.fullname}"
-            db.query_data(view_query)
-            logger.info(
-                "  Created %s view using %s temp table",
-                etl_view.fullname,
-                temp_table.fullname,
-            )
-
-    def update_bronze_pdr(self):
+    def do_bronze_queries(self):
         """
         @description
-            Update bronze tables
+            Run all bronze queries to process new + update events
         """
-        logger.info("update_bronze_pdr - Update bronze tables.")
+        logger.info("do_bronze_queries - start")
 
         # st_timestamp and fin_timestamp should be valid UnixTimeMS
         st_timestamp, fin_timestamp = self._calc_bronze_start_end_ts()
+        db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
 
-        for dataclass in _ETL_REGISTERED_LAKE_TABLES:
-            etl_func = dataclass.get_fetch_function()
-            etl_func(
-                path=self.ppss.lake_ss.lake_dir,
-                st_ms=st_timestamp,
-                fin_ms=fin_timestamp,
+        # if st_ts or fin_ts != ppss, then we must have records somewhere
+        is_first_run: bool = self._clamp_checkpoints_to_ppss is True or (
```

Comment:
I think you can have a separate function adnotated with `@property` on the class named is_first_run, i.e. self.is_first_run. Much easier to read logically.

Reply:
1. but it's not a "property" or a class variable... it's a check made before the ETL execution begins.
2. the other member variable _clamp_checkpoints_to_ppss, 
in addition to other checks (that are expensive) must be called in order to calculate this condition

perhaps i can move this somewhere else that makes more sense...

Reply:
thought about it a bit more...
1. `is_first_run` is specifically being calculated for the bronze_tables, based on `bronze_table` ETL requirements...
2. `silver_tables` and `gold_tables` may have other `is_first_run` logic/checks...

this doesn't make sense to generalize

Reply:
I would recommend separating the condition itself into a function. Call it is_first_bronze run if necessary, but it looks like a bunch of conditions that should have their own place and allow the reader of the code to quickly glimpse within a line "it is the first run", not get tangled in the conditionals. If needed, they can look at the function for details.

Reply:
```
but it looks like a bunch of conditions that should have their own place and allow the reader of the code to quickly glimpse within a line "it is the first run", not get tangled in the conditionals
```

(1) this check/logic is very specific to this routine
(2) it's an extremely simple condition
(3) every other process that can be abstracted and benefits from it, has been
(4) i'd like to expand on silver, gold, and other ETL workflows before abstracting things further

```
If needed, they can look at the function for details
```
abstracting everything also forces users to follow 5 threads to stitch the whole thing together

can we please expand on the system (silver, gold, other ETL) before trying to abstract everything?

Reply:
done